GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -78,23 +68,35 @@ private void Delete_Click(object sender, EventArgs e)
                 return;
             }
 
-            if (!AppSettings.DontConfirmDeleteUnmergedBranch)
+            bool areAllInReflog = selectedBranches.All(b => _reflogHashes.Contains(b.ObjectId.ToString()));
```

Comment:
This is O(n^2) operation, so it gets expensive with the size of selected branches and reflogs (the latter can be very large if a user actively rebases).
This is also trashing the memory with ToString.

Reply:
I think there is no performance problem here for multiple reason:
* in nearly every usage of this delete popup, `selectedBranches` will contain only one value so I don't think we will trash the memory with that 😉 (personally, in more than 13 years using GE and due to how the UI tend you to use it, I don't think I have deleted more than 1 branch at the same time one time....)
* O(n^2) is the worst case scenario where all the branches should be tested but if one is not in the reflog, the check is stopped.
* Whatever the way I try to test it, the time spend by the user to select the branches to delete is much more important than the check.
* I still have made some optimizations following your comment (cache the result to not do check twice, improved comparison instead of full ToString() and use of hashset) but the result time was already really imperceptible and is even more now: less than 2ms to check 90 branches with 1650 unique hashes in the reflog. 
* I built the worst case scenario where user want to delete 100 branches (and so take 1min to select them), with a reflog artificially grown to 56k hashes (and worst case, the one looked for are at the end), the check take 700ms.
 
So we can conclude that there is no performance problem here.